cv.py

```diff
@@ -62,89 +87,60 @@ def run_cv(self):
+            box = np.int0(box)
+            if area > 100:
+                rotated_boxes.append(RotatedRectangle(box, area, rot_angle))
 
-            cv2.drawContours(frame, [rotated_rect2.box], 0, (0, 0, 255), 2)
-            # cv2.putText(frame, "Rect2: " + str(rotated_rect2.rot_angle), (rotated_rect2.box[0][0], rotated_rect2.box[0][1]), cv2.FONT_HERSHEY_DUPLEX, 2, 255)
+        if(len(rotated_boxes) > 1):
+            rotated_rect1 = rotated_boxes[0]
+            rotated_rect2 = rotated_boxes[1]
+            if(len(rotated_boxes) > 2) :
+                rotated_rect3 = rotated_boxes[2]
+                rotated_rect1, rotated_rect2 = self.get_closest_rects(rotated_rect1, rotated_rect2, rotated_rect3)
 
-            top_point1 = max(rotated_rect1.box, key=lambda x: x[1])
-            top_point2 = max(rotated_rect2.box, key=lambda x: x[1])
 
-            bottom_point1 = min(rotated_rect1.box, key=lambda x: x[1])
-            bottom_point2 = min(rotated_rect2.box, key=lambda x: x[1])
+            r1_point1, r1_point2, r1_point3, r1_point4 = rotated_rect1.point1, rotated_rect1.point2, rotated_rect1.point3, rotated_rect1.point4
+            r2_point1, r2_point2, r2_point3, r2_point4 = rotated_rect2.point1, rotated_rect2.point2, rotated_rect2.point3, rotated_rect2.point4
 
-            top_dist = math.hypot(top_point2[0] - top_point1[0], top_point2[1] - top_point1[1])
-            bottom_dist = math.hypot(bottom_point2[0] - bottom_point1[0], bottom_point2[1] - bottom_point1[1])
+            self.pinX, self.pinY = self.calc_pin_pos(r1_point1, r1_point2, r1_point3, r1_point4, r2_point1, r2_point2, r2_point3, r2_point4)
 
-            # draws lines connecting the two top points of the targets and the two bottom points
-            cv2.line(frame, (top_point1[0], top_point1[1]), (top_point2[0], top_point2[1]), (255, 0, 0), 5)
-            cv2.line(frame, (bottom_point1[0], bottom_point2[1]), (bottom_point2[0], bottom_point2[1]), (255, 0, 0), 5)
+            # draws center circle
+            cv2.circle(frame, (self.pinX, self.pinY), 1, (255, 0, 0), thickness=5)
 
-        if (len(maxContours) < 2):
+        # if there are less than two rectangles, return -99, -1
+        if (len(rotated_boxes) < 2):
             cv2.imshow("Contours", mask)
             cv2.imshow("Frame", frame)
             cv2.waitKey(3)
-            return -1, -1
+            return -99, -1
 
-        rect1 = maxContours[0]
-        rect2 = maxContours[1]
-        rect3 = None
 
-        if len(maxContours) < 3:
-            rect3 = Rectangle(0, 0, 0, 0, 0)
-        else:
-            rect3 = maxContours[2]
 
-        oneRect = False
+        angle = self.calc_ang_deg(self.pinX)
+        self.angle_to_target = target
 
-        if rect3.area != 0:
-            rect1, rect2 = self.get_closest_rects(rect1, rect2, rect3)
+        # multiplying 1.5 yields consistently more accurage results
+        distance = 1.5 * self.calc_dist((rotated_rect1.height + rotated_rect2.height) / 2.0)
```

Comment:
This feels strange. Mathematically, multiplying by a constant won't make a random distribution more consistent. Do you have numbers on how much more "consistent" and "accurate" this was?